### PR TITLE
Leptos 0.7.0

### PR DESCRIPTION
**Pull request by ozpv**
Update to be compatible with leptos 0.7.0 
and updated some other dependencies of the examples as well as fixed a bug in the simple app one

I have only tested it on both of the examples but it seems to be working now

had to change the props from ```MaybeProp<TextProp>``` to ```Option<Oco<'static, str>>``` because of a trait implementation.
It's not implemented and says TODO, I found it commented out in the leptos main branch

### CODE REVIEW

**Comment by ozpv** [comment]
didn't realize it, but at first my changes weren't reactive, I fixed it now
and the examples work too
also realized that the problem isn't MaybeProp, so I replaced that and changed from text prop to string, it's just MaybeProp<TextProp>, String will have to do in the meantime

File: src/lib.rs
Comment (carloskiki) [inline src/lib.rs]: Why are we switching from `TextProp` to `String`?

**Review by carloskiki — CHANGES_REQUESTED** [review]
Actually, with new niceties surrounding attribute spreading I think we should remove the `class`, `style`, `width` and `height` props, to support [attribute spreading](https://book.leptos.dev/view/03_components.html?highlight=attributes#spreading-attributes-onto-components) instead.

File: src/lib.rs
Comment (ozpv) [reply src/lib.rs]: there was a trait impl error after the update

**Comment by ozpv** [comment]
Okay, so I removed the props for the attr spreading instead. one downside is now you have to wrap the icon prop value in curly braces, or it will error. Like:
`<Icon icon={i::BiGraphql} {..} width="2em" height="2em" style="color: green" />`
otherwise, setting defaults is easier now because attr spreading just seems to override the `.attr()` in the component

**Comment by ozpv** [comment]
I actually got around to testing the attr spreading in my project and it doesn't actually override the attrs set by the component. I thought it would because it makes sense that it would function that way, but I learned the reason the icons were the correct size in the examples I tested is because the width and height is set in the style attribute. 

Either we remove the default class (I don't think this is set anyway), width, and height attrs, or find a way to conditionally merge the classes and set the width and height if it isn't in the spread. or maybe you were thinking something else?

**Review by carloskiki — APPROVED** [review]
Awesome!

**Comment by carloskiki** [comment]
looks like leptos requires rustc `1.76.0` now, can you bump the `msrv` to make CI pass?

**Comment by ozpv** [comment]
I bumped the msrv, but there's still a problem.

I looked through the requests in the example apps as well as my own, and the spread functionality only replaces the attrs specified when the server isn't streaming HTML (when there's no SSR). Otherwise, the server is sending a svg tag with first the default `width` and `height` attrs, then the ones from the spread at the very end, which my browser (firefox) seems to truncate off rather than replace the original ones.

That means we're gonna need to find a way to replace the default attrs before anything is streamed (access the spread in the component somehow), or just revert back to including the optional `width`, `height`, and `class` as `MaybeProp<String>` in the meantime because of the current `MaybeProp<TextProp>` impl error

**Comment by carloskiki** [comment]
Got it. Thanks for your work!

Do you think this could be a bug related to `leptos`, or is this normal behavior?

**Comment by carloskiki** [comment]
Just remembered the msrv is hard coded in CI 🤦, if you want you can modify it in the PR otherwise don't worry about it I will fix it once I merge.

**Comment by ozpv** [comment]
> Do you think this could be a bug related to `leptos`, or is this normal behavior?

I don't believe it's a bug, however, I don't know why spreading wouldn't merge `class` and `style` attributes returned by the component, and replace `width` and `height` ones etc. as that functionality would make the most sense, rather than just appending it at the very end of the element. 

Whatever is in the WASM binary produced in the example does seem to do that for us, but when streaming certain HTML elements like this, no.

I think I might make a feature request or attempt a pr (I don't know proc macros very well) and explain this

> Just remembered the msrv is hard coded in CI 🤦, if you want you can modify it in the PR otherwise don't worry about it I will fix it once I merge.

as this might take a while if we don't come up with something in the meantime, I'll leave that up to you

**Comment by ozpv** [comment]
I ended up asking, and yes, it's working as it's supposed to.

Spreading should only be used when you want to **add extra attributes** and not replace them. So we will need the optional `width`, `height`, and `style` props (the `class` attr should be fine since there's no default needed).
It only happened to work in our CSR examples because the browser calls `setAttributes` which replaces it, while in SSR, it just adds it to the list of attributes.

That or we can just strip the defaults, but the icon will be invisible until you specify the `width` and `height` attr in a spread.
I think it's better to just keep 3 optional props, and remove the need for the class one.

**Comment by carloskiki** [comment]
v0.4.0 should be available on `crates.io`!

**Pull request merged**
